include/zephyr/drivers/spi/rtio.h

```diff
@@ -0,0 +1,114 @@
+/*
```

Comment:
I don't think this belongs to include/zephyr, it's not a public API as far as I know. Only spi drivers are meant to use it as it generalizes common parts. Thus its proper place should be in drivers/spi/ .

I don't know why it was done this way in i2c, it's anyway not uncommon to see variations in how things are done depending on the domain. For instance spi_rtio_copy ended in the public spi.h where instead, in i2c in ended up in i2c_rtio.h only via its signature (but public as well), even though in both case: it's meant to be used by drivers only.

Reply:
Well drivers may exist out of tree, I think this goes to a pr @dcpleung has. I generally believe the application and driver used apis should be public, just not necessarily in the same header. So far we have the one header for the application in include but that’s not very supportive of out of tree drivers.

Reply:
I saw daniel's PR, it's a good idea. So it could go into include/zephyr/drivers/internal/spi.h then

Reply:
It hasn’t been sorted out where that stuff should go exactly I don’t think, but yeah something along those lines

Reply:
> I saw daniel's PR, it's a good idea. So it could go into include/zephyr/drivers/internal/spi.h then

Can you provide a your rationale behind suggesting that path? I don't see how lumping _internal_ header files behind `drivers/internal/` is better than `drivers/<area>/*`.

In this case, specifically, I don't think we'll want to lump all-things SPI (`zephyr/drivers/internal/spi.h)` next to SPI RTIO (`zephyr/drivers/spi/rtio.h`).

Reply:
This is coming from https://github.com/zephyrproject-rtos/zephyr/pull/77361 and a related larger tree wide PR doing much the same

Reply:
> This is coming from #77361 and a related larger tree wide PR doing much the same

Thanks for the link. By looking at the diff + the comment below: It appears this goes along the same lines? 

> I would say that, in terms of a hierarchical tree, anything under drivers/serial and drivers/uart should be considered internal if they do not need to be exposed outside this "subtree". One can collapse this subtree into a single node and nothing else outside would be affected.